### PR TITLE
[FSDE-9221] Create backstop visual regresion test of data table

### PR DESCRIPTION
**Pull request by josecarlosgonzalezv**

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://07393ef93a431a46ecec428e89e8fe2f70debe3c-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/2/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://fc07c1f69c94deabc0e9c820a738de00a40dcb88-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: backstop-non-responsive.json [hunk 1]
```diff
@@ -1574,4 +1615,4 @@
   "asyncCompareLimit": 4,
   "debug": false,
   "debugWindow": false
-}
+}
```
Comment (evansito) [inline backstop-non-responsive.json:1618]: Newline at the end of file is missing

File: test/chi/components/data-table.pug [hunk 2]
```diff
@@ -0,0 +1,1347 @@
...
+              button.chi-button.-flat(aria-label='Page 5') 5
+              .chi-button.-flat(aria-hidden='true' disabled='') ...
+              button.chi-button.-flat(aria-label='Page 12') 12
+              button.chi-button.-icon.-flat(aria-label='Next page')
+                .chi-button__content
+                  i.chi-icon.icon-chevron-right(aria-hidden='true')
+          .chi-pagination__end
+            .chi-pagination__jumper
+              label.chi-pagination__label(for='jumper-input-17') Go to page:
+              input#jumper-input-17.chi-input(type='text')
```
Comment (evansito) [inline test/chi/components/data-table.pug:1347]: Same here

**Review by soujanreddy — APPROVED** [review]

**Review by evansito — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/3/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://af3f000658dff0aaf7887ee9ee83100217a9370a-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/4/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://419903d86dfdb98289c2b0dfbe88e7fabb1f21e9-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/5/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://e1ee0bfd3f673576e3ecf50f7f8b8883d07f3d81-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/6/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ab66cafad6918eceedb016d0c64f639235f06729-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

File: test/chi/components/data-table.pug [hunk 3]
```diff
@@ -0,0 +1,1347 @@
+---
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:1]: Can we break this down into separate files, like
_base.pug
_accordion.pug
_child.pug
etc

and import them into data-table/index.pug like:
include ./_base
include ./_accordion
include ./_accordion
?

File: test/chi/components/data-table.pug [hunk 4]
```diff
@@ -0,0 +1,1347 @@
...
+title: Data Table
+---
+
+- const sizes = ['xs', 'sm', 'md', 'lg', 'xl'];
+
+.test-base
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:7]:
```suggestion
.test-base.-w--100
```
Please, provide the class to all the test cases as currently, they render with a narrow width.

File: test/chi/components/data-table.pug [hunk 5]
```diff
@@ -0,0 +1,1347 @@
...
+      .chi-data-table__row
+        .chi-data-table__cell
+          div Name
+        .chi-data-table__cell
+          div ID
+        .chi-data-table__cell
+          div Last Login
+    .chi-data-table__body
+      .chi-data-table__row-empty
+      div No matches found. Please revise search criteria and try again.
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:89]:
```suggestion
        | No matches found. Please revise search criteria and try again.
```
Please test if this renders correctly as currently, the text appears outside the row.

File: test/chi/components/data-table.pug [hunk 6]
```diff
@@ -0,0 +1,1347 @@
...
+              button.chi-button.-flat(aria-label='Page 12') 12
+              button.chi-button.-icon.-flat(aria-label='Next page')
+                .chi-button__content
+                  i.chi-icon.icon-chevron-right(aria-hidden='true')
+          .chi-pagination__end
+            .chi-pagination__jumper
+              label.chi-pagination__label(for='jumper-input-3') Go to page:
+              input#jumper-input-3.chi-input(type='text')
+
+.test-hover
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:229]: This test case does not provide additional value as the rendered image does not show the hover state of the rows

Comment (janibolkvadze) [reply test/chi/components/data-table.pug:229]: Let's remove it.

File: test/chi/components/data-table.pug [hunk 7]
```diff
@@ -0,0 +1,1347 @@
...
+              input#jumper-input-6.chi-input(type='text')
+
+.test-selectable
+  h2 Selectable
+  .chi-data-table
+    .chi-data-table__head
+      .chi-data-table__row
+        .chi-data-table__cell.-selectable
+          .chi-checkbox
+            input#checkbox-ba1.chi-checkbox__input(type='checkbox')
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:443]:
```suggestion
            input#checkbox-ba1.chi-checkbox__input(type='checkbox' checked)
```

File: test/chi/components/data-table.pug [hunk 8]
```diff
@@ -0,0 +1,1347 @@
...
+        .chi-data-table__cell
+          div ID
+        .chi-data-table__cell
+          div Last Login
+    .chi-data-table__body
+      fieldset
+        .chi-data-table__row
+          .chi-data-table__cell.-selectable
+            .chi-radio
+              input#radio-ba2.chi-radio__input(type='radio' name='radios')
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:542]:
```suggestion
              input#radio-ba2.chi-radio__input(type='radio' name='radios' checked)
```

File: test/chi/components/data-table.pug [hunk 9]
```diff
@@ -0,0 +1,1347 @@
...
+              button.chi-button.-flat(aria-label='Page 2') 2
+              button.chi-button.-icon.-flat(aria-label='Next page')
+                .chi-button__content
+                  i.chi-icon.icon-chevron-right(aria-hidden='true')
+          .chi-pagination__end
+            .chi-pagination__jumper
+              label.chi-pagination__label(for='jumper-input-9') Go to page:
+              input#jumper-input-9.chi-input(type='text')
+
+.test-accordion-portal
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:686]: Portal test cases will be set up in a different way. Let's keep the file Lumen theme specific

File: test/chi/components/data-table.pug [hunk 10]
```diff
@@ -0,0 +1,1347 @@
...
+            .chi-data-table__cell
+              div ID
+            .chi-data-table__cell
+              div Last Login
+        .chi-data-table__body
+          .chi-data-table__row.-expanded.-xl
+            .chi-data-table__cell.-expandable
+              button.chi-button.-icon.-flat.-sm(aria-label='Expand row')
+                .chi-button__content
+                  i.chi-icon.icon-minus(aria-hidden='true')
```
Comment (janibolkvadze) [inline test/chi/components/data-table.pug:785]: This Expansion icon is from the Portal theme.
This and several following examples use portal markup. Please, keep the test page Lumen theme-specific.

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/7/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://39a502fbbe3e588d61e7c609fcb1c6421e4cd052-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/8/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://14de76ea4e888af1b74d982590b96278da67271c-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/9/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/10/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://9d91169de731b0cbffa4f82c33ebf536c4057a03-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/11/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://e0a377e8d260d375082108033db139a7ddc36eae-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/12/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://e711c68a7620e1e51fc6861aec7861fc44c64ecc-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/13/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/14/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://4655c8e60bc54f8187b81b003d2ac7f0d5a971be-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/15/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/16/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://6bb190ff97f98df9184ce38f035958e2f67d09fc-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/17/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://2fcf6a2eb2edcb07f6e3fa5669a518417affb7bf-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/19/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://30400e68199e7587df458e8e6ce5e1453ff068f8-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/20/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/21/. :x:

**Review by evansito — APPROVED** [review]

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/22/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/23/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://c033e6363d7a3e3015481c7f1919297aa14c3906-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/24/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/25/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/26/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://fef8ca7f622fd1781ff94d1568450000122943bf-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/28/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://7b36c321d037cf177a5cf82d740bd9c1274912e6-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/29/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://d7287caabccce9791bde82be80d626af60cfcd02-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/30/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://fc619f82778500db83f61b4df10728e65e4127c3-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/33/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/40/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/42/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/44/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://d37ff24762ac56d5331b7532d2cf605bcb4c0246-chi-pr.enterprise.ctl.io (external) and in https://nginx-pr-242-ux-chi.kubeodc-test.corp.intranet/ (internal)

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/47/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/49/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/50/. :x:

**Comment by SCMAUTO** [comment]
The CI pipeline did not run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-242/51/. :x:

**Pull request closed without merging**